### PR TITLE
New phase field fracture model that is anisotropic 

### LINKED ISSUE #8298: Anisotropic phase field fracture model
### Description of the enhancement or error report
Our current phase field fracture model assumes isotropic elastic properties.

### Rationale for the enhancement or information for reproducing the error
We use the current model to predict cracking in polycrystals, however we currently ignore the elastic anisotropy effect in the polycrystals, which will hurt accuracy. In addition, the isotropic model makes it difficult to determine the Jacobian.

### Identified impact
This is adding new capability and will not change any existing capability

### PR DESCRIPTION
**Pull request by tonkmr**
This implements a new phase field fracture model that works for anisotropic materials. This doesn't change the existing model, but eventually we won't need the other one. Closes #8298

### CODE REVIEW

File: modules/combined/tests/phase_field_fracture/crack2d_aniso.i [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    beta = b
+  [../]
+  [./solid_x]
+    type = StressDivergencePFFracTensors
+    variable = disp_x
+    component = 0
+    c = c
+  [../]
+  [./solid_y]
+    type = StressDivergencePFFracTensors
```
Comment (dschwen) [inline modules/combined/tests/phase_field_fracture/crack2d_aniso.i]: I may want to add support for this to the tensor mechanics master action.

File: modules/phase_field/include/kernels/ACPFFrac.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef ACPFFRAC_H
+#define ACPFFRAC_H
+/**
+ * Phase field based fracture model
+ * This kernel computes the residual and jacobian for bulk free energy contribution to c
+ * Refer to Formulation: Miehe et. al., Int. J. Num. Methods Engg., 2010, 83. 1273-1311 Equation 63
```
Comment (dschwen) [inline modules/phase_field/include/kernels/ACPFFrac.h]: The comment should be right above the class declaration.

File: modules/phase_field/include/kernels/ACPFFrac.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#include "JvarMapInterface.h"
+#include "DerivativeMaterialInterface.h"
+
+//Forward Declarations
+class ACPFFrac;
+
+template<>
+InputParameters validParams<ACPFFrac>();
+
+class ACPFFrac : public DerivativeMaterialInterface<JvarMapKernelInterface<Kernel> >
```
Comment (dschwen) [inline modules/phase_field/include/kernels/ACPFFrac.h]: I'd strongly prefer a more verbose class name. We have enough of cryptic classes already. `AllenCahnPFFracture` maybe? ...Wait, does `AC` even stand for Allen-Cahn? Or is it `AnisotropicCracking`? ... case and point.

File: modules/phase_field/include/kernels/ACPFFrac.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  /// Viscosity parameter ( visco -> 0, rate independent )
+  const MaterialProperty<Real> & _visco;
+
+  /// Free energy material properties
+  const MaterialProperty<Real> & _dFdc;
+  const MaterialProperty<Real> & _d2Fdc2;
+  const MaterialProperty<RankTwoTensor> & _d2Fdcdstrain;
+
+ private:
+
```
Comment (dschwen) [inline modules/phase_field/include/kernels/ACPFFrac.h]: remove

File: modules/tensor_mechanics/src/materials/ComputeLinearElasticPFFracStress.C [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    cfactor = 0.0;
+
+  // Compute eigenvectors and eigenvalues of ustress
+  RankTwoTensor eigvec;
+  std::vector<Real> eigval(LIBMESH_DIM);
+  _mechanical_strain[_qp].symmetricEigenvaluesEigenvectors(eigval, eigvec);
+
+  RankTwoTensor Ipos, Ineg, eigval_tensor, eye;
+  Ineg.zero();
+  eye.zero();
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ComputeLinearElasticPFFracStress.C]: We have a constructor to build an identity tensor.

**Review by dschwen — COMMENTED** [review]
This looks very nice. I can fix the minor things. Just give me some feedback on the name.

File: modules/combined/tests/phase_field_fracture/crack2d_aniso.i (hunk 1 above)
Comment (tonkmr) [inline modules/combined/tests/phase_field_fracture/crack2d_aniso.i]: That would be great.

**Comment by tonkmr** [comment]
The Jacobian matrix in this is not correct yet, so the solve gets slow when the order parameters gets near 1. @zhangsf2015 and I are working on it.

The other thing I would like to explore, and @dschwen I need your help on, would be making this work with any constitutive model. 

Essentially, I would like to be able to call any ComputeStress object, and then call this and have it MODIFY the existing stress material property and JacobianMult. So, we would have two materials modifying the same properties. I know this is problematic due to dependencies, but it would keep us from having to create a new version of the fracture ComputeStress object for each constitutive equation.

File: modules/phase_field/include/kernels/ACPFFrac.h (hunk 3 above)
Comment (tonkmr) [inline modules/phase_field/include/kernels/ACPFFrac.h]: AllenCahnPFFracture is fine

**Comment by tonkmr** [comment]
This can be merged before the issues I raised above are addressed. That is future work.

**Comment by dschwen** [comment]
@tonkmr, having two materials modify the same property is pretty much a no go. What you could do is have a 'base_name' on your material specific stress and couple the values into your fracture stress calculator and emit a non-prefixed stress and Jacobian mult.

**Comment by tonkmr** [comment]
@dschwen I've thought about that, but I'm not sure how it would work with incremental methods, like J2 plasticity. The old values of stress and strain would be unmodified and not include the impact of the crack, even though they actual stress calculation did.

**Comment by dschwen** [comment]
Good point. We could always have incremental methods take the old value from a different property!   But let's have an actual design discussion about this in a separate ticket. 

I'll fetch your branch, make the changes and open a new PR.

File: modules/phase_field/src/kernels/ACPFFrac.C [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+#include "MathUtils.h"
+
+template<>
+InputParameters validParams<ACPFFrac>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Kernel to compute bulk energy contribution to damage order parameter residual equation");
+  params.addParam<MaterialPropertyName>("l_name", "l", "Interface width");
+  params.addParam<MaterialPropertyName>("visco_name", "visco", "Viscosity parameter");
+  params.addParam<MaterialPropertyName>("gc_prop_var", "gc_prop", "Material property name with gc value");
```
Comment (dschwen) [inline modules/phase_field/src/kernels/ACPFFrac.C]: What is "gc"?

Comment (dschwen) [inline modules/phase_field/src/kernels/ACPFFrac.C]: critical fracture energy density?

**Comment by dschwen** [comment]
Closing in favor of #8483

**Pull request closed without merging**